### PR TITLE
ps_noise_sniffer example that uses pyadi-iio

### PR DESCRIPTION
**Pull request by andreeaipop**
This example has the same functionality as ps_noise_sniffer.py
from power_supply_noise_sniffer_example but it uses  pyadi-iio module which
provides a convenient way to access and control ADI hardware
from Python through existing IIO drivers.

Signed-off-by: Pop <Andreea.Pop@analog.com>

### CODE REVIEW

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 1]
```diff
@@ -0,0 +1,595 @@
...
+except:
+    print("Please install matplotlib")
+
+
+#Error
+ERROR = -1
+
+#Success
+SUCCESS = 0
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 2]
```diff
@@ -0,0 +1,595 @@
...
+
+FREQ_BAND = 40000000 #Frequency band  in Hz
+TXDAC_FREQ1=500000 #Frequency of test tone 1 in Hz
+TXDAC_FREQ2=10000000 #Frequency of test tone 2 in Hz
+
+BUFLEN=2**13 #buffer length
+
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+def check_connection(sdr):
+    if(sdr== ERROR):
+        txt1.config(fg = "red")
+        txt1.insert('1.0', "No Device Detected.\n\n")
+        return ERROR
+    else:
+        txt1.config(fg = 'black')
+        txt1.insert('1.0', "Device Connected.\n\n")
+        return SUCCESS
+#device setup
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Add new line between functions.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 4]
```diff
@@ -0,0 +1,595 @@
...
+    
+    #Compute center frequencies for each band
+        for i in range(1, nr_center_freq):
+            center_freq = np.append(center_freq, center_freq[i-1] + FREQ_BAND )
+    return center_freq, start_freq, stop_freq
+
+#Set hardware gain to the smallest value
+def set_hw_gain(center_freq):
+    
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 5]
```diff
@@ -0,0 +1,595 @@
...
+        time.sleep(0.2)
+        hw_gain_arr=np.append(hw_gain_arr, sdr.rx_hardwaregain) 
+    
+    min_hw_gain = min(hw_gain_arr)
+    
+    #set manual hw gain
+    min_hw_gain = min(hw_gain_arr)
+    sdr.gain_control_mode = 'manual'
+    min_hw_gain=float(min_hw_gain.replace(' dB', ""))    
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 6]
```diff
@@ -0,0 +1,595 @@
...
+    #set manual hw gain
+    min_hw_gain = min(hw_gain_arr)
+    sdr.gain_control_mode = 'manual'
+    min_hw_gain=float(min_hw_gain.replace(' dB', ""))    
+    
+    
+    return min_hw_gain
+    
+
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 7]
```diff
@@ -0,0 +1,595 @@
...
+def load_gain_calib():
+    global gain_freq
+    gain_freq = []
+    
+    with open('gain_calib.csv') as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            gain_freq.append(tuple((row['Frequency(MHz):'], row['Magnitude(dB):'])))
+
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 8]
```diff
@@ -0,0 +1,595 @@
...
+
+def load_filter(sdr):  
+        
+    #enable/disable FIR filter for TX and RX 
+    if(filter_en.get() == True):
+        txt1.insert(tk.END, "FIR Filter enabled.\n\n")
+        sdr.filter="LTE20_MHz.ftr"
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 9]
```diff
@@ -0,0 +1,595 @@
...
+        txt1.config(fg = 'red')
+        txt1.delete('1.0', tk.END)
+        txt1.insert('1.0', "Please insert valid start/stop frequencies.")
+        return ERROR
+    else:
+        txt1.config(fg = "black")
+        txt1.insert(tk.END, "Start frequency set to: " + str(start_freq/1e6) + " MHz\n\n")
+        txt1.insert(tk.END, "Stop frequency set to: " + str(stop_freq/1e6) + " MHz\n\n")
+        return SUCCESS    
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 10]
```diff
@@ -0,0 +1,595 @@
...
+                    newband = np.abs(np.fft.fft(rxvals_iq)) 
+                    #apply averaging
+                    if(k != 0):
+                        avgband = (newband) * 1/k + avgband * (k-1) / k;
+                    else:
+                        avgband = newband
+            else:     
+                for j in range(5):
+                    x = sdr.rx()
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 11]
```diff
@@ -0,0 +1,595 @@
...
+            root.update_idletasks()
+            root.update()  
+                   
+            if(btn_text.get() == "Start" and sweep_en.get() == False):
+                break
+            
+        if(btn_text.get() == "Start" or sweep_en.get() == True):
+            break       
+        avg_step += 1
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 12]
```diff
@@ -0,0 +1,595 @@
...
+    #first sweep for getting hardware gain   
+    min_hw_gain=set_hw_gain(center_freq)
+    txt1.insert(tk.END, "Minimun hardware gain: " + str(min_hw_gain) + ' dB\n')
+    get_plot_data(sdr, center_freq, min_hw_gain)
+    
+    btn_export.config(state = tk.NORMAL)
+    
+    sweep_en.set(False)
+    
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 13]
```diff
@@ -0,0 +1,595 @@
...
+        return
+    else:
+        btn_text.set("Stop")
+        btn_start.config(bg = "red")
+        btn_export.config(state = tk.DISABLED)
+        
+    #Clear GUI Content
+    clr_content()
+  
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 14]
```diff
@@ -0,0 +1,595 @@
...
+    #Check connection
+    if(check_connection(sdr) == ERROR):
+       btn_text.set("Start")
+       btn_start.config(bg = "lime green")
+       return
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 15]
```diff
@@ -0,0 +1,595 @@
...
+    time.sleep(0.1)
+    
+    load_gain_calib()
+    
+    #first sweep for getting hardware gain   
+    min_hw_gain=set_hw_gain(center_freq)
+    txt1.insert(tk.END, "Minimun hardware gain: " + str(min_hw_gain) + ' dB\n')
+    while(btn_text.get() == "Stop"):
+        get_plot_data(sdr, center_freq, min_hw_gain)
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 16]
```diff
@@ -0,0 +1,595 @@
...
+    
+    root.mainloop()
+
+#main function
+def main():
+    gui()
+
+main()
+
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 17]
```diff
@@ -0,0 +1,595 @@
...
+    txt1.insert(tk.END, "Minimun hardware gain: " + str(min_hw_gain) + ' dB\n')
+    while(btn_text.get() == "Stop"):
+        get_plot_data(sdr, center_freq, min_hw_gain)
+    
+    
+    btn_export.config(state = tk.NORMAL)
+    
+    btn_text.set("Start")
+    btn_start.config(bg = "lime green")
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

File: pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py [hunk 18]
```diff
@@ -0,0 +1,595 @@
...
+    try:
+        int(avg_nr.get())
+        return SUCCESS
+    except:
+        txt1.config(fg = "red")
+        txt1.insert(tk.END, "Please insert valid average number.")
+        btn_text.set("Start")
+        btn_start.config(bg = "lime green")
+        return ERROR
+
```
Comment (amiclaus) [inline pluto/python/pyadi-iio_power_supply_noise_sniffer_example/ps_noise_sniffer_pyadi_iio.py]: Extra line. Please remove.

**Comment by andreeaipop** [comment]
changelog v1->v2:
 - add header
- remove extra lines

**Pull request merged**